### ISSUE TITLE
Shell task (typescript watch) doesn't start with integrated shell set to cmd.exe

### ISSUE THREAD

**Issue by jeffrson** [Microsoft/vscode#64896]
- VSCode Version: 1.29.1 / 1.30.0
- OS Version: Windows 10 1809

I have a typescript watch task:
```
{
            "args": [
                "--watch",
                "-p",
                "\"${workspaceFolder}/subfolder/tsconfig.json\""
            ],
            "command": "tsc",
            "problemMatcher": "$tsc-watch",
            "type": "shell",
            "isBackground": true
}

```

The task works well without `terminal.integrated.shell.windows` or when set to "powershell.exe". But when I use "cmd.exe" the task hangs "somewhere". Actually it starts, because it complains if tsconfig.json is missing.

Does this issue occur when all extensions are disabled?: Yes

**Comment by jeffrson** [comment]
Strange thing: when I run procmon (sysinternals) in parallel, the task works as expected - race condition?

**Comment by alexr00** [comment]
Do you have a repo I can clone that has this issue? This is not a task I'd expect to work anywhere since `tsc` isn't a command. I would expect the command to be `node`.

**Comment by jeffrson** [comment]
So here is the short demo movie.
1. Starts powershell as default. No prompt. Then I click inside the terminal (you can see a circle flashing) and press "Enter". Only then the prompt appears.
2. Start the watch task which appears rather quickly - two lines of output.
3. Change shell to cmd.exe - start new terminal. The terminal is launched with PowerShell, but asks for permission to run cmd. Permission granted, terminal restarted - it's cmd now.
4. Start the watch task. Nothing happens :-(
Furthermore you can see that I often need to scroll in order to look at the output.

![vscode_cmd](https://user-images.githubusercontent.com/9551280/49992790-582ede80-ff85-11e8-96c6-ec74e0b73a3f.gif)

**Comment by alexr00** [comment]
I still can't reproduce, but your video looks a lot like #57803 and you're on the right Windows build for that (1809) If it's #57803, then the task is actually running, the terminal output is just messed up. Can you start the task, then open on of your watched typescript files, make a change, and save it? You should see the task terminal finally output what you expect.

**Comment by alexr00** [comment]
I wish I had better news, but based on #57803 you just have to wait until your PC updates past 1809.
